src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -10,8 +11,106 @@ import {
 import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance, useAllowancesData } from '../hooks';
+
+const EditModal = (isOpen, onClose, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const {
+    allowancesList,
+  } = useAllowancesData();
+
+  const allowances = allowancesList.filter(a => a.id === allowance.id);
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+
+  const initialFormState = {};
+  const [form, setForm] = useState(initialFormState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setForm(prev => ({ ...prev, [name]: value }));
+  };
+
+  const onSubmit = () => {
+    // todo: make sure that validation is correct
+    setAdditionalTimeError(!form['extra-time-mins']);
```

Comment:
TODO: I think I need to update the handling of validation. When I try to insert a string instead of a number, it doesn't allow submission (as expected). But when I try to add a negative number, I get this error attached (syntax error). So going to investigate that. But any feedback on validation handling is appreciated (I just pulled this from the add allowance modal).
![Screenshot 2024-08-15 at 11 04 05 AM](https://github.com/user-attachments/assets/e4728e5f-707f-4d73-acb7-907177544e32)

Reply:
Created this ticket to handle this: https://2u-internal.atlassian.net/browse/COSMO-446